azurelinuxagent/common/utils/networkutil.py

```diff
@@ -200,21 +248,27 @@ def __execute_cmd(cmd):
 
     @staticmethod
     def add_iptables_rules(wait, dst_ip, uid):
-
+        # The order in which the below rules are added matters for the ip table rules to work as expected
         AddFirewallRules.__raise_if_empty(dst_ip, "Destination IP")
         AddFirewallRules.__raise_if_empty(uid, "User ID")
 
-        accept_rule = AddFirewallRules.get_iptables_accept_command(wait, AddFirewallRules.__ACCEPT_COMMAND, dst_ip, uid)
+        accept_rule_dns_tcp_rule = AddFirewallRules.get_iptables_accept_dns_tcp_request_command(wait, AddFirewallRules.get_insert_command(),dst_ip)
+        AddFirewallRules.__execute_cmd(accept_rule_dns_tcp_rule)
+
+        accept_rule = AddFirewallRules.get_iptables_accept_command(wait, AddFirewallRules.get_append_command(), dst_ip, uid)
```

Comment:
can we be consistent and either append or insert all the rules?

Reply:
There is no issue using append everywhere when it comes to iptable rules.However,there is an issue when it comes to the Distros that persist the firewall rules using firewalld

For example in RHEL , since it persist the rules we have to use insert in order to maintain the order. Therefore an insert for the new rule in case of persist firewall scenario using firewalld is a must since the older agent has already set the older ACCEPT and DROP rule

I considered using Insert in both iptable rule and firewalld . I that case insert needs an index to be provided for rule number 2 and 3

Example: 
['iptables', '-t', 'security', '-A', 'OUTPUT', **'2'** '-d', '168.63.129.16', '-p', 'tcp', '-m', 'owner', '--uid-owner', '0', '-j', 'ACCEPT', '-w']

But since the other commands such as DELETE and CHECK need not require an index , the function will have an additional parameter that will be used only for insert . I felt it is a bit clumsy.
 
Therefore I settled with using append for all iptable rules and insert for the new rule for firewalld.

I can change it to insert everywhere and write logic to use the index only in case of insert if that is suggested.